### PR TITLE
fix(packges): add mac build targets for tiflow

### PR DESCRIPTION
**Pull request by wuhuizuo**
What changes for tiflow
===
1. add `darwin` as support target os.
2. add os conditional steps for tiflow darwin targets, we use nvm to provide different nodejs versions.

### CODE REVIEW

**Comment by wuhuizuo** [comment]
> @wuhuizuo:
> 
> > **I have already done a preliminary review for you, and I hope to help you do a better job.**
> 
> ## Review for "fix(packges): add mac build targets for tiflow"
> ### Summary
> This pull request makes two changes to the `tiflow` package:
> 
> 1. Adds `darwin` as a supported target OS.
> 2. Adds OS conditional steps for `tiflow` darwin targets, using `nvm` to provide different node versions.
> 
> ### Potential Problems
> * The pull request does not mention what version of macOS is supported. It is recommended to add the minimum supported version of macOS.
> * The installation process for `node` and `yarn` is hardcoded to a specific version. It is recommended to make these versions configurable through environment variables or command-line flags.
> * The installation of `nvm` is hard-coded to install in `/usr/local/nvm`. It is recommended to make this configurable as well.
> * The `build_package_artifacts.sh` file has been changed to include the `build_binaries` function. This function is not used anywhere else in the file. It is recommended to either remove this function or use it in the script.
> 
> ### Fixing Suggestions
> * Add the minimum supported version of macOS.
> * Make the installation process for `node` and `yarn` configurable through environment variables or command-line flags.
> * Make the installation of `nvm` configurable.
> * Remove the `build_binaries` function or use it in the script.

The AI misunderstanding the changes.

File: packages/packages.yaml.tmpl [hunk 1]
```diff
@@ -166,7 +164,26 @@ components:
...
+                if [ ! -s "$NVM_DIR/nvm.sh" ]; then
+                  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/${NVM_VERSION}/install.sh | bash
+                fi
+                . $NVM_DIR/nvm.sh
+                nvm install ${NODE_VERSION}
+                nvm use v${NODE_VERSION}
+                nvm alias default v${NODE_VERSION}
+
+                node --version && npm --version
+                npm install -g yarn
```
Comment (lijie0123) [inline packages/packages.yaml.tmpl:184]: should toolchain installing script be here? install toolchain every time seems slow. How about just write the dependencies?
Besides, there is no llvm/gcc installing script for tikv/tiflash

Comment (wuhuizuo) [reply packages/packages.yaml.tmpl:184]: It's tiflow!

Comment (wuhuizuo) [reply packages/packages.yaml.tmpl:184]: tikv, tiflash is not ready.

nvm install will skip if the version installed, so do not worry about it.

File: packages/scripts/build-package-images.sh.tmpl [hunk 2]
```diff
@@ -83,7 +83,7 @@ main() {
     {{- end }}
     {{- end }}
     {{- end }}
-
+
```
Comment (wuhuizuo) [inline packages/scripts/build-package-images.sh.tmpl:86]:
```suggestion
```

**Pull request merged**